starknet-core/src/types/block.rs

```diff
@@ -1,23 +1,29 @@
-use super::{super::serde::deserialize_h256_from_hex, ConfirmedTransactionReceipt, Transaction};
+use super::{
+    super::serde::{deserialize_h256_from_hex, deserialize_option_h256_from_hex},
+    ConfirmedTransactionReceipt, Transaction,
+};
 
 use ethereum_types::H256;
 use serde::Deserialize;
 
 pub enum BlockId {
     Hash(H256),
     Number(u64),
+    Pending,
 }
 
 #[derive(Debug, Deserialize)]
 pub struct Block {
-    #[serde(deserialize_with = "deserialize_h256_from_hex")]
-    pub block_hash: H256,
-    pub block_number: u64,
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_option_h256_from_hex")]
```

Comment:
A very minor nit - the `default` and `deserialize_with` could be written on a single line, in one attribute. Same applies to the `state_root` below. I'm fine with merging as it tho.

Reply:
Yes ser changing it